handler.go

```diff
@@ -516,6 +512,49 @@ func populateTmpDir(r *http.Request, rmDir bool, dirpath string) (string, *gz.Er
 	return dirpath, nil
 }
 
+// extractFilepath extracts the full filename from the Content-Disposition header.
+// If it's not found, it returns the default Filename from the given multipart.FileHeader
+func extractFilepath(fh *multipart.FileHeader) (string, error) {
+	_, params, err := mime.ParseMediaType(fh.Header["Content-Disposition"][0])
```

Comment:
`fh.Header["Content-Disposition"][0]` may panic if the `Content-Disposition` header is not present.